### ISSUE TITLE
[Intermittent] Mozilla VPN is crashing after swipe-killing the app while the VPN is ON

### ISSUE THREAD

**Issue by data-sync-user** [mozilla-mobile/mozilla-vpn-client#9224]
**Version/build found in:**

* 2.21.0 (2.202403072206);

**Reproducible on:**

* iOS 17.3.1;

**Prerequisites:**

* Have the Mozilla VPN app installed and opened;
* Have a Mozilla account with an active subscription;
* Be signed in; 

**Steps to reproduce:**

1. Turn the VPN ON
1. Wait for about 1-2 minutes, then swipe-kill the app.
1. Reopen the app.

**Expected result:**

* The activation screen is displayed.

**Actual result:**

* The app is crashing. The crash report screen was not displayed - only the modal from the OS.

**Notes:**

* See attached logs;

┆Issue is synchronized with this [Jira Bug](https://mozilla-hub.atlassian.net/browse/VPN-6243)
┆Reporter: Raluca Monica Ilban

**Comment by data-sync-user** [comment]
➤ Bianca Hidecuti commented:

Hi Matt Cleinman,

As Raluca mentioned, I was able to reproduce it only once, while verifying the previous crash ticket (VPN-5766).

Today I was unable to reproduce it while using the latest build - will give it another go in the next few days and update the ticket as we go.

Thanks.

**Comment by data-sync-user** [comment]
➤ Bianca Hidecuti commented:

I was able to reproduce the crash again while using Mozilla VPN 2.21.0 (2.202403122206) on iOS 17.3.1, after changing the location a few times and swipe-killing / reopening the app.

The crash reporter screen was still not displayed, only the OS modal. Time of crash: 14:55.

Attaching logs as well. 

[^14.55 mozillavpn.log]

**Comment by data-sync-user** [comment]
➤ Matt Cleinman commented:

I don’t think this would be iOS 17 related, but I can’t say for sure. 

Nothing is jumping out to me from those logs, unfortunately. (I do see some stuff in ConnectionHealth shortly before the crashes, but that file hasn’t been changed in months.) Additionally, it doesn’t seem like we got anything from Sentry for today’s crash.

If we get additional logs, sentry data, or steps that reliably reproduce the problem, I’ll dig into it. Without any of those, I’m afraid it’s going to be a bit difficult to figure out exactly what is happening here.

**Comment by data-sync-user** [comment]
➤ Raluca Monica Ilban commented:

Hi Matt Cleinman ,

We've encountered a crash today, using Mozilla VPN 2.22.0 (2.202404102207) build (3796) on iOS 15, after a Reset VPN action. The crash reporter screen was not displayed.
Crash time 18:21.Attaching logs.
Please let us know if this should be tracked separately. Thank you!

[^mozillavpn_18_21.log]

**Comment by data-sync-user** [comment]
➤ Matt Cleinman commented:

Hi Raluca Monica Ilban  - If it happens every time after the Reset VPN action or if it’s happening on other platforms, let’s put it in a new ticket. If it’s only on iOS and only sometimes, let’s keep it here. Thanks!

**Comment by data-sync-user** [comment]
➤ Raluca Monica Ilban commented:

Hi Matt Cleinman , it only happened once yesterday on iOS. I tried multiple times after that and was not able to reproduce the crash again.

**Comment by data-sync-user** [comment]
➤ Matt Cleinman commented:

Thanks. No need for a new ticket then.

**Comment by data-sync-user** [comment]
➤ Raluca Monica Ilban commented:

We've encountered a crash , using Mozilla VPN 2.24.0 (2.202406032209) on iOS 15, after swipe killing the app while the VPN was ON. 
Aprox crash time 13:28. Attaching logs.

[^mozillavpn.log]
